synapse/config/server.py

```diff
@@ -165,7 +165,6 @@ def generate_ip_set(
     "http",
     "metrics",
     "manhole",
-    "replication",
```

Comment:
We talked briefly about whether providing `replication` should show a deprecation warning instead of error. This has been deprecated for almost two years, so seems reasonable to just remove.

Reply:
Agreed

Reply:
76dcd63464564952c2ac25bd30ca6f87e03d8fbd adds an error message.